### ISSUE TITLE
Limits for headless

### ISSUE THREAD

**Issue by amin212121** [mollie/Shopware6#327]
Hi, I use your plugin for my APP which was builded like PWA.

But I have a bug. When user buy something over 2000$ and choose Credit card like payment method, after call handlePayment(from Shopware SDK) your plugin return error url

My questions: 
1) Do your plugins has like "hide methods" for PWA (for NO PWA I have seen it has)?
2) If no then how I can avoid this, mb u have some best practices ?

**Comment by boxblinkracer** [comment]
Hi

so basically there is no full headless support at the moment
but I'm about to start doing it next week 

what I think is happening though is, that the Mollie request (creating of the payment) leads to an error because the credit card limit might be high enough in your mollie account.
i think the default maximum is 1500 for credit card payments (you need to talk to your mollie accountant).

still...if a payment fails, it will always lead to an error.
so in the storefront, there is usually a redirect to the editOrder with the information to retry the payment.
...at the moment i dont know how it would behave with headless (but next week i will)

in other words: i think the headless/pwa (or your app) needs to also provide a handling for errors that happen during a payment


i hope this helps a bit

**Comment by amin212121** [comment]
@boxblinkracer any updates?
And what I thought maybe it's possible use Shopware rule builder and hide a payment method when amount of cart over 1500, what do u think or it has unexpected bahave?

**Comment by boxblinkracer** [comment]
Hi @amin212121 
so the mollie limits have already been done to automatically hide those payments in the api
so they should already be headless ready (although only tested with a storefront channel and the store-api that is available in the swagger doc).

the error what you have is a standard error if a payment fails, this is not because the card is declined, but because your merchant account isnt configured properly for such a high number.
in that case, the error handling as is will be used, which leads to errors in the log.

you can of course use any rules or anything that allows you to handle such scenarios. this is not really connected to the plugin, just plain shopware availability rules.
the mollie limits itself are (at the moment) only using the values from Mollie as thresholds, I'm not sure if it is even possible to use dynamic ranges for custom merchant settings...

and regarding the support for headless
it is going quite well, i had 1-2 days last week, but do of course need more time to have everything completely headless

i hope this helps you :)

**Comment by amin212121** [comment]
@boxblinkracer thank you for your work :)

**Comment by boxblinkracer** [comment]
released with v3.0
